prov/rxd/src/rxd_atomic.c

```diff
@@ -0,0 +1,451 @@
+/*
+ * Copyright (c) 2013-2018 Intel Corporation. All rights reserved
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <sys/uio.h>
+
+#include "ofi_iov.h"
+#include "rxd.h"
+
+static void rxd_ioc_to_iov(const struct fi_ioc *ioc, struct iovec *iov,
+			   size_t count, size_t size)
+{
+	int i;
+	for (i = 0; i < count; i++) {
+		iov[i].iov_base = ioc[i].addr;
+		iov[i].iov_len = ioc[i].count * size;
+	}
+}
```

Comment:
Doesn't this function exist in common code somewhere already?

Reply:
If it does, I couldn't find it. I know I wrote one of these in the smr provider. I can move that into common code if it's not already there and use it for both. Do you know where it would go if it did exist or if I were to move it there?

Reply:
I don't see a common function either based on searches.  I would put these in ofi_iov.h.